ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule.hs

```diff
@@ -312,6 +324,33 @@ mkPeers h as =
 pointSchedule :: [Tick] -> NonEmpty PeerId -> Maybe PointSchedule
 pointSchedule ticks nePeerIds = (`PointSchedule` nePeerIds) <$> nonEmpty ticks
 
+peerStates :: PeerId -> [(DiffTime, SchedulePoint)] -> [(DiffTime, Peer NodeState)]
+peerStates peerId =
+  fmap (second (Peer peerId . NodeOnline)) .
+  scanl' step (0, zero)
+  where
+    step (_, z) =
+      fmap $ \case
+        ScheduleTipPoint tip -> z {tip = TipPoint tip}
+        ScheduleHeaderPoint h -> z {header = HeaderPoint h}
+        ScheduleBlockPoint b -> z {block = BlockPoint b}
+
+    zero = AdvertisedPoints {
+      tip = TipPoint TipGenesis,
+      header = HeaderPoint (TestHeader genesisBlock),
+      block = BlockPoint genesisBlock
+    }
+
+    genesisBlock = (firstBlock 0) {tbSlot = 0}
+
+fromSchedulePoints :: Map PeerId [(DiffTime, SchedulePoint)] -> Maybe PointSchedule
+fromSchedulePoints peers = do
+  peerIds <- nonEmpty (Map.keys peers)
+  pointSchedule (mkTick <$> states) peerIds
+  where
+    mkTick (t, s) = Tick s (Just t)
+    states = sortOn fst (uncurry peerStates =<< Map.toList peers)
```

Comment:
```suggestion
    states = mergeOn fst $ concat [peerStates p sch | (p, sch) <- Map.toList peers]
```
Am I the only one who finds the list comprehension easier to read?
Note also the change from `sortOn` to `mergeOn`.